### PR TITLE
Fixes ZEN-30428

### PR DESCRIPTION
**Pull request by zsalinas**
Add audit logging to the devicemanagement router. Most complaints were around the lack of audit logging for Maintenance Window actions, however we were missing audit logging for User Command and Role assignments as well, this adds logging for all of those actions.

### CODE REVIEW

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

File: Products/Zuul/routers/devicemanagement.py [hunk 1]
```diff
@@ -41,6 +43,7 @@ def deleteMaintWindow(self, uid, id):
         """
         facade = self._getFacade()
         data = facade.deleteMaintWindow(uid, id)
+        audit('UI.MaintenanceWindow.Delete', id)
```
Comment (StevePC) [inline Products/Zuul/routers/devicemanagement.py]: maintenance window IDs are not globally unique.  you should also log the uid or parent so we know which MW of this name is being deleted

File: Products/Zuul/routers/devicemanagement.py [hunk 2]
```diff
@@ -72,6 +75,7 @@ def editMaintWindow(self, params):
         """
         facade = self._getFacade()
         data = facade.editMaintWindow(params)
+        audit('UI.MaintenanceWindow.Edit', params['id'])
```
Comment (StevePC) [inline Products/Zuul/routers/devicemanagement.py]: for edit operations, do we not include *what* changed in the audit log?  I'm sure I've seen that somewhere...

File: Products/Zuul/routers/devicemanagement.py (hunk 1 above)
Comment (zsalinas) [reply Products/Zuul/routers/devicemanagement.py]: good point, i'll add that.

File: Products/Zuul/routers/devicemanagement.py (hunk 2 above)
Comment (zsalinas) [reply Products/Zuul/routers/devicemanagement.py]: i know that we do that for at least some of them. the way the maintenance window parameters are passed in i'm not sure if there is a good way to specifically glean the changes out but i'll check. good catch.

**Comment by dlysak** [comment]
retest this

**Comment by dlysak** [comment]
retest this

**Comment by jpeacock-zenoss** [comment]
retest this

**Comment by jpeacock-zenoss** [comment]
Since this PR was opened by Zack, the PR build won't run until Zack sets his membership in the Zenoss organization public.

**Comment by dlysak** [comment]
retest this

**Pull request merged**